rfcs/0017-tx-valid-since/0017-tx-valid-since.md

```diff
@@ -15,97 +15,246 @@ This RFC suggests adding a new consensus rule to prevent a cell to be spent befo
 
 ## Summary 
 
-Transaction adds a new `u64` type field `valid_since`, which prevents the transaction to be mined before a certain point, the highest bit of `valid_since` is `type flag`, the other 63 bits represent the actual value `V`, the consensus to validate this field described as follow:
+Transaction input adds a new `u64` (unsigned 64-bit integer) type field `valid_since`, which prevents the transaction to be mined before an absolute or relative time.
 
-* ignore this validate rule if all bits of `valid_since` are 0.
-* `V` represents a block number when `type flag` is 0, the validation MUST fail if `tip.block_number < V`.
-* `V` represents a block timestamp when `type flag` is 1, the timestamp is represented as `V * 512` seconds, the validation MUST fail if `tip.timestamp < V * 512`.
+The highest 8 bits of `valid_since` is `flags`, the remain `56` bits represent `value`, `flags` allow us to determine behaviours:
+* `flags & (1 << 7)` represent `absolute_flag`.
+* `flags & (1 << 6)` represent `metric_flag`.
+    * `valid_since` use a block based lock-time if `metric_flag` is `0`, `value` can be explained as a block number or a relative number.
+    * `valid_since` use a time based lock-time if `metric_flag` is `1`, `value` can be explained as a block timestamp(unix time) or a relative seconds.
+* other 6 `flags` bits remain for other use.
 
-Otherwise, the transaction validation SHOULD continue.
+The consensus to validate this field described as follow:
+* iterate inputs, and validate each input by following rules.
+* ignore this validate rule if all 64 bits of `valid_since` are 0.
+* check `metric_flag`:
+    * the lower 56 bits of `valid_since` represent block number if `metric_flag` is `0`.
+    * the lower 56 bits of `valid_since` represent block timestamp if `metric_flag` is `1`.
+* check `absolute_flag`:
+    * consider field as absolute lock time if `absolute_flag` is `0`:
```

Comment:
consider field as absolute lock time if `absolute_flag` is `0`?

`0` not `1` ?

Reply:
Yes